src/frontend/mame/ui/miscmenu.cpp

```diff
@@ -51,6 +51,35 @@ menu_bios_selection::menu_bios_selection(mame_ui_manager &mui, render_container
 {
 }
 
+int menu_bios_selection::current_bios(device_t *device)
+{
+	const char *val;
+	std::string bios;
+	if (device->owner())
+	{
+		const char *slot_option_name = device->owner()->tag() + 1;
+		bios = machine().options().slot_option(slot_option_name).bios();
+		val = bios.empty() ?  nullptr : bios.c_str();
+	}
+	else
+	{
+		val = machine().options().value("bios");
+	}
+	if (val)
+	{
+		for (romload::system_bios const &bios : romload::entries(device->rom_region()).get_system_bioses())
```

Comment:
You’ve got a local `bios` here that shadows another local called `bios` declared earlier in the function and still in scope.  That’s a bad practice.

Reply:
Good catch, shadowing is fixed